Improve DB audit logging documentation (#19534)
pingcap/docs: tidb-cloud/tidb-cloud-auditing.md

To enable audit logging for AWS, take the following steps:

> Note:
>

> > Please do not enable object lock on AWS S3. If object lock is enabled, TiDB Cloud will be unable to push audit log files to S3.

Reviewer: ```suggestion
> Please do not enable object lock on AWS S3 bucket. If object lock is enabled, TiDB Cloud will be unable to push audit log files to S3.
```

> > Please do not enable object lock on AWS S3 bucket. If object lock is enabled, TiDB Cloud will be unable to push audit log files to S3.

Reviewer 2: ```suggestion
> Do not enable object lock on the AWS S3 bucket. Enabling object lock will prevent TiDB Cloud from pushing audit log files to S3.
```
